app/database/group_ancestor_store.go

```diff
@@ -17,7 +17,7 @@ func (s *GroupAncestorStore) UserAncestors(user *User) *DB {
 }
 
 // ManagedByUser returns a composable query for getting all the groups_ancestors rows for groups
-// that are descendants of groups managed by the user
+// that are descendants of groups managed by the user (the result may contain duplicates)
```

Comment:
I don't get how you could get duplicate due to managers while `group_managers` is unique on (`group_id`,`manager_id`), you join on `group_id` and you filter on `manager_id`.

Reply:
A user can be a manager of the group and some of its parents at the same time. So as we join group_managers to `groups_ancestors.ancestor_group_id`, we can end up with duplicates.

Reply:
> About
> 
> > Actually, there is an additional small (or not so small) error/inconsistency:
> > You have defined in group_managers that manager_id is a group (not just a user)... but in this PR, in ManagedByUser, you do check that the user himself is manager but not part of a group who is manager. So the current state is at least inconsistent. I'll discuss this with Mathias to decide if enforcing that the manager is an end user is ok (would be the easiest change and the safest in terms of perf)
> 
> We are going to make it correctly so make sure that "the user is a descendant of a group manager of the one of the ancestor of the group" for `ManagedByUser`
> Make sure you have tests for each place where it is changed but don't creates tests for each service using this function.

fixed

Reply:
> A user can be a manager of the group and some of its parents at the same time. So as we join group_managers to `groups_ancestors.ancestor_group_id`, we can end up with duplicates.

(about the initial query) I still don't get it. Yes of course, you can have duplicate `child_group_id` in the result and `ancestor_group_id` in the result. But you do not have duplicate `(ancestor_group_id, child_group_id)` so you do not have duplicate "groups_ancestors rows". 
Actually, if you have a table groups_ancestors at first with no duplicate, and you JOIN with something which has no duplicate on the `group_id` (each user can be manager of a given group only once), you cannot have multiplication of rows from the LEFT clause. So you cannot have duplicate of groups_ancestors.